### PR TITLE
[16.0][IMP] auth_oidc: prompt for account on AAD login

### PR DESCRIPTION
**Pull request by CRogos**
Added the possibility to add additional parameters to the auth_link. 
When using the AzureAD auth, this gives the ability to set the "prompt" parameter to "select_account" which allowes to display the  select account dialog during the login process. This is helpful, when you have multiple AzureAD accounts.

image

### CODE REVIEW

File: auth_oidc/controllers/main.py [hunk 1]
```diff
@@ -43,7 +44,17 @@ def list_providers(self):
...
+                # append provider specific auth link params
+                params_upd = literal_eval(provider["auth_link_params"])
+                params.update(params_upd)
+
                 # auth link that the user will click
+                provider["auth_link"] = "%s?%s" % (
+                    provider["auth_endpoint"],
+                    url_encode(params),
+                )
+
```
Comment (sbidoul) [inline auth_oidc/controllers/main.py]:
```suggestion
```

This is already done just below.

File: auth_oidc/views/auth_oauth_provider.xml [hunk 2]
```diff
@@ -18,6 +18,7 @@
             <field name="validation_endpoint" position="after">
                 <field name="token_endpoint" />
                 <field name="jwks_uri" />
+                <field name="auth_link_params" />
```
Comment (sbidoul) [inline auth_oidc/views/auth_oauth_provider.xml]: Maybe place this field next to the `auth_endpoint` field from the parent view?

**Review by sbidoul — COMMENTED** [review]
Nice!

File: auth_oidc/views/auth_oauth_provider.xml (hunk 2 above)
Comment (sbidoul) [reply auth_oidc/views/auth_oauth_provider.xml]: What I meant here is something like this, so the auth_link_params field is next to the auth_endpoint field on screen, as they are so closely related:

```
<field name="auth_endpoint" position="after">
  <field name="auth_link_params" />
</field>
```

**Review by sbidoul — APPROVED** [review]
Still a minor ergnonomy issue, otherwise looks good.

**Comment by sbidoul** [comment]
Ah, can you rebase and resolve the conflict on the CONFIGURE readme fragment?
There were rendering issues with the images, and while fixing that I took the opportunity to convert them to markdown.

**Review by lucuswolfius — APPROVED** [review]
Code lgtm, nice work

**Comment by CRogos** [comment]
@sbidoul can you merge?

**Comment by dreispt** [comment]
@CRogos Can you resolve the conflict please?

**Comment by CRogos** [comment]
@dreispt it is rebased. can you merge?

**Comment by sbidoul** [comment]
@CRogos do you plan to port this to 17?

**Pull request merged**